src/main/java/edu/kit/ipd/crowdcontrol/objectservice/database/operations/AnswerRatingOperations.java

```diff
@@ -22,21 +23,33 @@
 public class AnswerRatingOperations extends AbstractOperations {
     private final CalibrationOperations calibrationOperations;
     private final WorkerCalibrationOperations workerCalibrationOperations;
+    private final ExperimentOperations experimentOperations;
 
     public AnswerRatingOperations(DSLContext create, CalibrationOperations calibrationOperations,
-                                  WorkerCalibrationOperations workerCalibrationOperations) {
+                                  WorkerCalibrationOperations workerCalibrationOperations, ExperimentOperations experimentOperations) {
         super(create);
         this.calibrationOperations = calibrationOperations;
         this.workerCalibrationOperations = workerCalibrationOperations;
+        this.experimentOperations = experimentOperations;
     }
 
     /**
      * inserts a new answer into the DB
      * @param answerRecord the record to insert
      * @return the resulting record
+     * @throws IllegalArgumentException if the experiment the answers is referring to is not existing
+     * @throws IllegalStateException if the worker is not allowed to submit more answers
      */
-    public AnswerRecord insertNewAnswer(AnswerRecord answerRecord) {
+    public AnswerRecord insertNewAnswer(AnswerRecord answerRecord) throws IllegalArgumentException, IllegalStateException {
         answerRecord.setIdAnswer(null);
+        ExperimentRecord experiment = experimentOperations.getExperiment(answerRecord.getExperiment())
+                .orElseThrow(() -> new IllegalArgumentException("Illegal experiment-value in rating record."));
```

Comment:
Error message might be misleading. Shouldn't it say "Illegal experiment-id in answer record."